ostd/src/task/mod.rs

```diff
@@ -204,7 +209,8 @@ impl TaskOptions {
         /// all task will entering this function
         /// this function is mean to executing the task_fn in Task
         extern "C" fn kernel_task_entry() -> ! {
-            // See `switch_to_task` for why we need this.
+            // See `switch_to_task` for why we need them.
+            processor::set_prev_task_state();
             crate::arch::irq::enable_local();
```

Comment:
I suggest combining these two lines into a single hook function. This prevents leaking implementation details from the `processor` module to the `task` module. Good for readability and maintainability.

```rust
// file: task/mod.rs

        extern "C" fn kernel_task_entry() -> ! {
            processor::on_task_entry();

            ...
        }

// file: task/processor.rs

/// A callback function that MUST be called on the very beginning of the entry point
/// of a kernel thread.
pub(crate) fn on_task_entry() {
    processor::set_prev_task_state();
    crate::arch::irq::enable_local();
}
```